gcp_variant_transforms/testing/integration/run_tests_common.py

```diff
@@ -16,74 +16,93 @@
 
 It provides common functions and classes for both run_vcf_to_bq_tests
 (integration test script for vcf_to_bq pipeline) and run_preprocessor_tests
-(integration test script for vcf_to_bq preprocessor pipeline).
+(integration test script for vcf_to_bq_preprocess pipeline).
 """
 
-from abc import abstractmethod
 from typing import Dict, List  # pylint: disable=unused-import
 import argparse  # pylint: disable=unused-import
 import json
 import os
 import time
 
 from googleapiclient import discovery
+from oauth2client.client import GoogleCredentials
 
-DEFAULT_IMAGE_NAME = 'gcr.io/gcp-variant-transforms/gcp-variant-transforms'
+_DEFAULT_IMAGE_NAME = 'gcr.io/gcp-variant-transforms/gcp-variant-transforms'
+_DEFAULT_ZONES = ['us-west1-b']
 
 
 class TestCaseFailure(Exception):
   """Exception for failed test cases."""
   pass
 
 
-class TestCase(object):
-  """Test case that holds information to run in pipelines API."""
-
-  def __init__(self, test_name, project):
-    # type: (str, str) -> None
-    self._name = test_name
-    self._project = project
-
-  def get_name(self):
-    return self._name
-
-  def run(self, context):
-    service = discovery.build(
-        'genomics', 'v1alpha2', credentials=context.credentials)
-    # The following pylint hint is needed because `pipelines` is a method that
-    # is dynamically added to the returned `service` object above. See
-    # `googleapiclient.discovery.Resource._set_service_methods`.
-    # pylint: disable=no-member
-    request = service.pipelines().run(body=self._pipelines_api_request)
+def form_pipeline_api_request(project,  # type: str
+                              logging_location,  # type: str
+                              image,  # type: str
+                              scopes,  # type: List[str]
+                              pipeline_name,  # type: str
+                              script_path,  # type: str
+                              args  # type: List[str]
+                             ):
+  # type: (...) -> Dict
+  return {
+      'pipelineArgs': {
+          'projectId': project,
+          'logging': {'gcsPath': logging_location},
+          'serviceAccount': {'scopes': scopes}
+      },
+      'ephemeralPipeline': {
+          'projectId': project,
+          'name': pipeline_name,
+          'resources': {'zones': _DEFAULT_ZONES},
+          'docker': {
+              'imageName': image,
+              'cmd': ' '.join([script_path] + args)
+          }
+      }
+  }
+
+
+def run(pipelines_api_request):
+  # type: (Dict) -> Dict
+  service = discovery.build(
```

Comment:
This is the `service` object I mentioned in the other related comment; not in this PR but add a TODO not to recreate it on every call but instead create/keep only one copy as a class attribute and reuse.